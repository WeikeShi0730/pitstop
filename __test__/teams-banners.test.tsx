import { render, RenderResult } from "@testing-library/react";
import TeamsBanners from "../components/teams-banners.component";

let documentBody: RenderResult;
describe("<TeamsBanners />", () => {
  beforeEach(() => {
    const teams = [
      {
        id: "test",
        name: "test",
        teamBackgrounds: [],
        teamBanner: "test",
      },
    ];
    documentBody = render(<TeamsBanners teams={teams} />);
  });
  it("shows content in <TeamsBanners />", () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
