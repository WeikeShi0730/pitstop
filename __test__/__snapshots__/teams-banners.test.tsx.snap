// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamsBanners /> shows content in <TeamsBanners /> 1`] = `
<body>
  <div>
    <div
      class="flex flex-col justify-center m-auto my-10 relative w-3/5"
    >
      <div
        style="transform: perspective(1000px) rotateX(0deg) rotateY(0deg) scale3d(1, 1, 1);"
      >
        <div
          class="relative w-full h-32 rounded-lg my-3 transition-all duration-200 ease-in-out"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="test background image"
              class="object-cover rounded-md"
              data-nimg="fill"
              decoding="async"
              src="test"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
          <div
            class="flex justify-center items-center absolute w-full h-full opacity-0 hover:opacity-95 hover:backdrop-blur-sm rounded-md"
          >
            <a
              class="text-sm md:text-base bg-slate-700 text-slate-100 w-48 p-3 text-center rounded-md shadow-slate-700 shadow-md hover:bg-slate-600 hover:text-white transition-all ease-in-out duration-200"
              href="/test"
            >
              Shop 
              test
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
